src/GRA.Domain.Service/EmailManagementService.cs

```diff
@@ -104,5 +108,43 @@ public async Task TokenUnsubscribe(string token)
 
             await _userRepository.SaveAsync();
         }
+
+        public async Task<DataWithCount<ICollection<EmailTemplate>>>
+            GetPaginatedEmailTemplateListAsync(BaseFilter filter)
+        {
+            if (HasPermission(Permission.ManageBulkEmails))
+            {
+                return await _emailTemplateRepository.PageAsync(filter);
+            }
+            else
+            {
+                _logger.LogError($"User {GetClaimId(ClaimType.UserId)} doesn't have permission to view the email template list.");
```

Comment:
Use structured logging, not interpolated strings

Reply:
Fixed and noted